File: slycalendarview/src/main/java/ru/slybeaver/slycalendarview/SlyCalendarDialog.java
package ru.slybeaver.slycalendarview;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Calendar;
import java.util.Date;

import ru.slybeaver.slycalendarview.listeners.DialogCompleteListener;

/**
 * Created by psinetron on 03/12/2018.
 * http://slybeaver.ru
 */
public class SlyCalendarDialog extends DialogFragment implements DialogCompleteListener {


    private SlyCalendarData slyCalendarData = new SlyCalendarData();
    private Callback callback = null;

    public SlyCalendarDialog setStartDate(@Nullable Date startDate) {
        slyCalendarData.setSelectedStartDate(startDate);
        return this;
    }

    public SlyCalendarDialog setEndDate(@Nullable Date endDate) {
        slyCalendarData.setSelectedEndDate(endDate);
        return this;
    }

    public SlyCalendarDialog setSingle(boolean single) {
        slyCalendarData.setSingle(single);
        return this;
    }

    public SlyCalendarDialog setFirstMonday(boolean firsMonday) {
        slyCalendarData.setFirstMonday(firsMonday);
        return this;
    }

    public SlyCalendarDialog setCallback(@Nullable Callback callback) {
        this.callback = callback;
        return this;
    }

    public SlyCalendarDialog setTimeTheme(@Nullable Integer themeResource) {
        slyCalendarData.setTimeTheme(themeResource);
        return this;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, R.style.SlyCalendarDialogStyle);
    }

    @Nullable
    public Date getCalendarFirstDate() {
        return slyCalendarData.getSelectedStartDate();
    }

    @Nullable
    public Date getCalendarSecondDate() {
        return slyCalendarData.getSelectedEndDate();
    }


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        SlyCalendarView calendarView = (SlyCalendarView) getActivity().getLayoutInflater().inflate(R.layout.slycalendar_main, container);
        calendarView.setSlyCalendarData(slyCalendarData);
        calendarView.setCallback(callback);
        calendarView.setCompleteListener(this);
        return calendarView;
    }

    @Override
    public void complete() {
        this.dismiss();
    }


    public interface Callback {
        void onCancelled();

        void onDataSelected(Calendar firstDate, Calendar secondDate, int hours, int minutes);
    }


    public SlyCalendarDialog setBackgroundColor(Integer backgroundColor) {
        slyCalendarData.setBackgroundColor(backgroundColor);
        return this;
    }

    public SlyCalendarDialog setHeaderColor(Integer headerColor) {
        slyCalendarData.setHeaderColor(headerColor);
        return this;
    }

    public SlyCalendarDialog setHeaderTextColor(Integer headerTextColor) {
        slyCalendarData.setHeaderTextColor(headerTextColor);
        return this;
    }

    public SlyCalendarDialog setTextColor(Integer textColor) {
        slyCalendarData.setTextColor(textColor);
        return this;
    }

    public SlyCalendarDialog setSelectedColor(Integer selectedColor) {
        slyCalendarData.setSelectedColor(selectedColor);
        return this;
    }

    public SlyCalendarDialog setSelectedTextColor(Integer selectedTextColor) {
        slyCalendarData.setSelectedTextColor(selectedTextColor);
        return this;
    }


}

File: slycalendarview/src/main/java/ru/slybeaver/slycalendarview/SlyCalendarDialog.kt
package ru.slybeaver.slycalendarview

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import ru.slybeaver.slycalendarview.listeners.DialogCompleteListener
import java.util.*

/**
 * Created by psinetron on 27/05/20
 * http://slybeaver.ru
 */
class SlyCalendarDialog : DialogFragment(), DialogCompleteListener {

    private var slyCalendarData = SlyCalendarData()
    private var callback: Callback? = null

    interface Callback {
        fun onCancelled()
        fun onDataSelected(firstDate: Calendar?, secondDate: Calendar?, hours: Int, minutes: Int)
    }

    override fun complete() {
        dismiss()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NORMAL, R.style.SlyCalendarDialogStyle)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val calendarView = layoutInflater.inflate(R.layout.slycalendar_main, container) as SlyCalendarView
        calendarView.setSlyCalendarData(slyCalendarData)
        calendarView.setCallback(callback)
        calendarView.setCompleteListener(this)
        return calendarView
    }


    fun setStartDate(startDate: Date?): SlyCalendarDialog? {
        slyCalendarData.selectedStartDate = startDate
        return this
    }

    fun setEndDate(endDate: Date?): SlyCalendarDialog? {
        slyCalendarData.selectedEndDate = endDate
        return this
    }

    fun setSingle(single: Boolean): SlyCalendarDialog? {
        slyCalendarData.single = single
        return this
    }

    fun setFirstMonday(firsMonday: Boolean): SlyCalendarDialog? {
        slyCalendarData.firstMonday = firsMonday
        return this
    }

    fun setCallback(callback: Callback?): SlyCalendarDialog? {
        this.callback = callback
        return this
    }

    fun setTimeTheme(themeResource: Int?): SlyCalendarDialog? {
        slyCalendarData.timeTheme = themeResource
        return this
    }

    fun getCalendarFirstDate(): Date? {
        return slyCalendarData.selectedStartDate
    }

    fun getCalendarSecondDate(): Date? {
        return slyCalendarData.selectedEndDate
    }


    fun setBackgroundColor(backgroundColor: Int?): SlyCalendarDialog? {
        slyCalendarData.backgroundColor = backgroundColor
        return this
    }

    fun setHeaderColor(headerColor: Int?): SlyCalendarDialog? {
        slyCalendarData.headerColor = headerColor
        return this
    }

    fun setHeaderTextColor(headerTextColor: Int?): SlyCalendarDialog? {
        slyCalendarData.headerTextColor = headerTextColor
        return this
    }

    fun setTextColor(textColor: Int?): SlyCalendarDialog? {
        slyCalendarData.textColor = textColor
        return this
    }

    fun setSelectedColor(selectedColor: Int?): SlyCalendarDialog? {
        slyCalendarData.selectedColor = selectedColor
        return this
    }

    fun setSelectedTextColor(selectedTextColor: Int?): SlyCalendarDialog? {
        slyCalendarData.selectedTextColor = selectedTextColor
        return this
    }

    fun showTime(show: Boolean): SlyCalendarDialog?  {
        slyCalendarData.showTime = show
        return this
    }
}